pkg/metadata/metadata.go

```diff
@@ -54,10 +54,25 @@ func IncomingHeader(ctx context.Context) *Header {
 }
 
 func NewContextWithOutgoingMetadata(ctx context.Context, md Metadata) context.Context {
-	return context.WithValue(ctx, metadataKey{}, ctxMD{md: md})
+	if len(md) == 0 {
+		return ctx
+	}
+	m, ok := ctx.Value(metadataKey{}).(ctxMD)
+	if ok && m.md != nil {
+		m.md = maps.Clone(m.md)
```

Comment:
this removed the only way to reset the outgoing metadata. we shouldn't change this